### ISSUE TITLE
OBS fails to launch if Python partially removed with multiple installs

### ISSUE THREAD

**Issue by FireCulex** [obsproject/obs-studio#9487]
### Operating System Info

Windows 11

### OBS Studio Version

29.1.3

### OBS Studio Log URL

https://obsproject.com/logs/TLO5Q6VOUjFXQ2fi

### Expected Behavior

For Obs to Launch 

### Current Behavior

Nothing

### Steps to Reproduce

1. Uninstall Python
2. Attempt to Launch OBS
3.
...

### Anything else we should know?

Eventually I figured out removing this line got OBS to launch again:
[Python]
Path64bit=C:/Users/Culex/AppData/Local/Programs/Python/Python310

from global.ini. I had forgotten I had removed 3.10 previously. Ideally a warning or popup should occur, or something in the log as opposed to simply nothing.

**Comment by WizardCM** [comment]
So to confirm, it visually doesn't open? Or it opens and immediately closes?

**Comment by Fenrirthviti** [comment]
Also, a log where OBS successfully loads would be useful as well here. It seems to fail during the load plugins stage, which is an odd place for python being missing to cause issues.

**Comment by FireCulex** [comment]
> So to confirm, it visually doesn't open? Or it opens and immediately closes?

Nothing is shown on the screen. The log is generated but nothing useful. No crash. 
The easiest way to try see this for yourself is to add
```
[Python]
Path64bit=C:/Users/Culex/AppData/Local/Programs/Python/Python310
```
to %appdata%\obs-studio\global.ini

**Comment by FireCulex** [comment]
I apologize it appears I was mistaken in my assessment. I added `%appdata\Local2` and it loaded perfectly fine. Then I added `C:/Users/Culex/AppData/Local/Programs/Python/Python310` and OBS locked up. Attempting to terminate it in task manager resulted in a "Access is Denied" I had to reboot. 

Upon further inspection Python310 was only partially uninstalled. python.exe, pythonw.exe,python3.dll, python310.dll were removed, however etc Lib/site-packages Scripts/jupyter related and share are still present.

I loaded up windbg and tried it again.
```
ModLoad: 00007ffa`78730000 00007ffa`78b5c000   C:\Users\Culex\AppData\Local\Programs\Python\Python38\python38.dll
Fatal Python error: failed to get the Python codec of the filesystem encoding
```
Well thats odd, I'm not selecting python38 but python310. I renamed python38 to python38.old and selected python310 and now it works as in it doesn't lockup?? I renamed python38.old back to python38 and selected python38 and that works perfectly as well. 

So to recap when I select a partially uninstalled python310 with a working python38 installed, bad things happen.

**Comment by RytoEX** [comment]
On Windows 11, if I install Python 3.10 to:
```
C:\Users\User\AppData\Local\Programs\Python\Python310
```

And then I:
1. Set up Scripting in OBS to load that Python.
2. Set up the demo url-text.py with a text source.
3. Close OBS.
4. Uninstall Python 3.10.
5. Open OBS.

Everything works fine. OBS starts up and displays a notice in the Scripts window's Python Settings tab that, "Python not currently loaded".

However, if I have both Python 3.8 and Python 3.10 installed and I uninstall only Python 3.10, OBS fails to start at all (the process appears and then disappears in Task Manager). I do not get any weird process/UI hangs or lockups, but OBS will not finish startup.

**Comment by RytoEX** [comment]
I've figured out why an older version of Python might be loaded if the specified version of Python is not found. In this loop, we continuously decrement the specified minor version of Python until a version of Python is found.
https://github.com/obsproject/obs-studio/blob/29.1.3/deps/obs-scripting/obs-scripting-python-import.c#L80-L94

However, we do not update the Python path with whatever path is ultimately found, so we call Py_SetPythonHome with an incorrect path, which causes a fatal error:
```
Fatal Python error: failed to get the Python codec of the filesystem encoding
```
This causes OBS to crash silently.
